src/main/java/org/jabref/gui/linkedfile/LinkedFileDialogController.java

```diff
@@ -31,21 +31,23 @@ public class LinkedFileAddDialogController extends BaseDialog<LinkedFile> {
 
     private LinkedFilesEditDialogViewModel viewModel;
     private final LinkedFile linkedFile;
+    private final boolean isEditMode;
 
-    public LinkedFileAddDialogController() {
-        this.linkedFile = new LinkedFile("", "", "");
+    public LinkedFileDialogController(LinkedFile linkedFile, boolean isEditMode) {
+        this.linkedFile = linkedFile != null ? linkedFile : new LinkedFile("", "", "");
+        this.isEditMode = isEditMode;
```

Comment:
Instead of making it so complicated, you can overload the constructor. One will take a LinkedFile for editing, another will not take any argument, for adding file. Invoke them accordingly.
This way, 1. you won't have to pass null values 2. won't have to pass an additional editMode flag.